StockCount/View/EditViewController.swift

```diff
@@ -24,26 +26,71 @@ class EditViewController: CustomNavBarViewController, UITextViewDelegate
     var keyboardToolBar : KeyboardToolBar?
     private let appleKeyboardIdentifier = ["en_US@hw=Automatic;sw=QWERTY", "en_US@sw=QWERTY;hw=Automatic"]
     
+    let volumeControl = MPVolumeView(frame: CGRect(x: 0, y: 0, width: 120, height: 120))
+    var isForceVolumeChange = false
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
         updateTopButtons(isDoneVisible: false)
         
         //Keyboard Notification
-        NotificationCenter.default.addObserver(self, selector: #selector(self.keyboardNotification(notification:)), name: NSNotification.Name.UIKeyboardWillChangeFrame, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(self.keyboardNotification(notification:)), name: UIResponder.keyboardWillShowNotification, object: nil)
         
         keyboardToolBar = KeyboardToolBar(frame: CGRect(x: 0, y: 0, width: self.view.frame.size.width, height: 50))
         keyboardToolBar?.delegate = self
         txtView.inputAccessoryView = keyboardToolBar
         
         editController.readFile(fileName)
+        
+        view.addSubview(volumeControl)
+    }
+    override func viewDidLayoutSubviews() {
+        volumeControl.frame = CGRect(x: -120, y: -120, width: 100, height: 100)
     }
     
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+    }
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         txtView.becomeFirstResponder()
+        
+        AVAudioSession.sharedInstance().addObserver(self, forKeyPath: "outputVolume", options: .new, context: nil)
+        
+        setVolume(0.5)
```

Comment:
why `0.5`?

Reply:
volume is in 0~1.
And we have to change volume when it reaches to max or min to detect next volume change.
For example, if volume is max and user click volume up button, then it doesn't change volume, so we can't detect volume button click.
So I made volume to 0.5 when it reaches 0 or 1.
0.5 can be any value bigger than 0 and smaller than 1. no need to have to be 0.5
But decided middle of it.
Please let me know if you have any other opinion.

Reply:
That does not "sound" right ;-) It's kind of weird to have an app that changes your volume. Isn't a better way, like setting 0.1 when min, and 0.9 when max?

Reply:
- changed code to set volume 0.1 or 0.9.